tempesta_fw/sock_srv.c

```diff
@@ -946,6 +959,90 @@ tfw_cfgop_lookup_sg_cfg(const char *name)
 	return NULL;
 }
 
+static int
+tfw_cfgop_srv_create_health(const char *hname, TfwServer *new_srv,
+			    TfwServer *orig_srv)
+{
+	TfwSrvHealth *health;
+	TfwServer *srv;
+	size_t size = 0;
+	bool orig_hm = false;
+
+	if (orig_srv)
+		orig_hm = test_bit(TFW_SRV_B_HMONITOR,
+				   (unsigned long *)&orig_srv->hm_flags);
+
+	/*
+	 * Nothing to do if the same server with the same
+	 * hmonitor, or the same server without hmonitor (and
+	 * new hmonitor is not specified), or new server
+	 * without hmonitor at all.
+	 */
+	if ((hname && orig_hm && tfw_apm_hm_srv_eq(hname, orig_srv))
+	    || (!hname && orig_srv && !orig_hm)
+	    || (!hname && !orig_srv))
+		return 0;
+
+	if (hname)
+		size = strlen(hname) + 1;
+	health = kzalloc(sizeof(TfwSrvHealth) + size, GFP_KERNEL);
+	if (!health) {
+		TFW_ERR_NL("can't allocate memory for server health entry\n");
+		return -ENOMEM;
+	}
+
+	if (hname) {
+		if (orig_hm)
+			health->flags |= TFW_SRV_HEALTH_DISABLE;
+		health->flags |= TFW_SRV_HEALTH_ENABLE;
+		health->name = (char *)(health + 1);
+		memcpy(health->name, hname, size);
+		srv = orig_srv ? : new_srv;
+	} else {
+		srv = orig_srv;
+		health->flags |= TFW_SRV_HEALTH_DISABLE;
+	}
+
+	INIT_LIST_HEAD(&health->list);
+	health->srv = srv;
+	list_add_tail(&health->list, &tfw_cfg_health_lst);
+
+	return 0;
+}
+
+static int
+tfw_cfg_srv_set_health(void)
+{
+	TfwSrvHealth *hth;
```

Comment:
There're some common shortcuts that are understood in a specific way. For instance, nth usually mean n-th consecutive number, or something like that. Of course you can name variables any way you like in a small function, but it's best if the name resemble an unambiguous meaning.

There's a number of "rules" for creating short names, one of which is removing all vowels: `default` -> `dflt`, etc. With that, it would be `health` -> `hlth`. Or just leave it `health`, it's sufficiently short in this particular case. Or even just `h` - as long as it can't be confused with h-th number or something. :-)

Reply:
This code has been redesigned.